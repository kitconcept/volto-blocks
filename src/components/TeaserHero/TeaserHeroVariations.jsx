import React from 'react';
import { Icon } from '@plone/volto/components';
import { Button } from 'semantic-ui-react';
import heroTopSVG from '@plone/volto/icons/hero.svg';
import heroLeftSVG from '../../icons/hero-left.svg';
import heroRightSVG from '../../icons/hero-right.svg';

const TeaserHeroVariations = ({ variation, onChangeBlock, data, block }) => {
  function onSetVariation(variation) {
    onChangeBlock(block, {
      ...data,
      variation,
    });
  }

  return (
    <div>
      <Button.Group>
        <Button
          icon
          basic
          aria-label="Top"
          onClick={() => onSetVariation('top')}
          active={data.variation === 'top' || !data.variation}
        >
          <Icon name={heroTopSVG} size="24px" />
        </Button>
      </Button.Group>
      <Button.Group>
        <Button
          icon
          basic
          aria-label="Left"
          onClick={() => onSetVariation('left')}
          active={data.variation === 'left'}
        >
          <Icon name={heroLeftSVG} size="24px" />
        </Button>
      </Button.Group>
      <Button.Group>
        <Button
          icon
          basic
          aria-label="Right"
          onClick={() => onSetVariation('right')}
          active={data.variation === 'right'}
        >
          <Icon name={heroRightSVG} size="24px" />
        </Button>
      </Button.Group>
    </div>
  );
};

export default TeaserHeroVariations;
