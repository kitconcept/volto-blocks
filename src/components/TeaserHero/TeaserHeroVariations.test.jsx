import React from 'react';
import renderer from 'react-test-renderer';
import TeaserHeroVariations from './TeaserHeroVariations';

test('renders a teaser hero variations component', () => {
  const component = renderer.create(
    <TeaserHeroVariations
      variation="left"
      onChangeBlock={() => {}}
      data={{
        '@type': 'teaserHero',
        href: '/de/a-page/news1',
        variation: 'left',
        padded: true,
      }}
      block="1c58b5a2-26fb-4112-832b-fad88af1ae0e"
    />,
  );
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});
