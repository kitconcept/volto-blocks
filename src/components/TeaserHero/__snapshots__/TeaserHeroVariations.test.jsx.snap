// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a teaser hero variations component 1`] = `
<div>
  <div
    className="ui buttons"
  >
    <button
      aria-label="Top"
      className="ui basic icon button"
      onClick={[Function]}
    >
      <svg
        className="icon"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "currentColor",
            "height": "24px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
    </button>
  </div>
  <div
    className="ui buttons"
  >
    <button
      aria-label="Left"
      className="ui active basic icon button"
      onClick={[Function]}
    >
      <svg
        className="icon"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "currentColor",
            "height": "24px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
    </button>
  </div>
  <div
    className="ui buttons"
  >
    <button
      aria-label="Right"
      className="ui basic icon button"
      onClick={[Function]}
    >
      <svg
        className="icon"
        dangerouslySetInnerHTML={
          Object {
            "__html": undefined,
          }
        }
        onClick={null}
        style={
          Object {
            "fill": "currentColor",
            "height": "24px",
            "width": "auto",
          }
        }
        viewBox=""
        xmlns=""
      />
    </button>
  </div>
</div>
`;
